src/lib/y2storage/filesystems/blk_filesystem.rb

```diff
@@ -406,6 +401,33 @@ def preferred_mount_by
 
         Filesystems::MountByType.best_for(self, mount_bys)
       end
+
+      private
+
+      # Convenience method for checking if the given STRING is a well formed UUID
+      #
+      # @param spec [String] content of the first column of an /etc/fstab entry
+      # @return [Boolean]
+      def uuid?(spec)
+        /^UUID="(.*)"/ =~ spec || /^UUID='(.*)'/ =~ spec || /^UUID=(.*)/ =~ spec
+      end
+
+      # Convenience method for checking if the given STRING is a well formed label
+      #
+      # @param spec [String] content of the first column of an /etc/fstab entry
+      def label?(spec)
+        /^LABEL="(.*)"/ =~ spec || /^LABEL='(.*)'/ =~ spec || /^LABEL=(.*)/ =~ spec
+      end
+
+      # @return [Boolean]
+      def match_uuid?
```

Comment:
I don't like the idea of using `#last_match` here if the matching actually happened in another method. This forces you to call `#match_uuid?` always right after calling `#uuid?` and only if that call succeeded. Too fragile.

Reply:
Yep, have the same opinion

Reply:
I would have expected something like this:

```ruby
def match_fstab_spec?(spec)
  return true if match_fstab_uuid?(spec)
  return true if match_fstab_label?(spec)

  named_device = devicegraph.find_by_any_name(spec)
  blk_devices.include?(named_device)
end
```

Reply:
After modifying the Regexp we do not need any of these changes at all.